### PR TITLE
Remove redux dependencies from JSON Forms

### PR DESCRIPTION
**Pull request by TheZoker**
Removes redux as a direct dependency. 
- Redux is removed from the material package (not needed anyway)
- The missing types have been replaced with own types
- `jsonformsRecducer` was removed from the core package (not used anyway)
- All material tests have been rewritten to run without redux
- `react-redux` was moved from the `peerDepedencies` to the `optionalPeerDependencies`
- The `jsonFormsReduxContext` is now available under `@jsonforms/react/redux`
```
import { jsonFormsReduxContext } from `@jsonforms/react/lib/redux`;
```

Related issue: https://github.com/eclipsesource/jsonforms/issues/1537

### CODE REVIEW

File: packages/core/src/reducers/index.ts [hunk 1]
```diff
@@ -78,20 +76,6 @@ export {
...
-    core: coreReducer,
-    renderers: rendererReducer,
-    cells: cellReducer,
-    config: configReducer,
-    uischemas: uischemaRegistryReducer,
-    defaultData: defaultDataReducer,
-    i18n: i18nReducer,
-    ...additionalReducers
-  });
-
```
Comment (sdirix) [inline packages/core/src/reducers/index.ts:94]: We need to keep the `jsonformsReducer` as they are needed when using JSON Forms with Redux, see for example the React seed or the example apps.

**Review by sdirix — CHANGES_REQUESTED** [review]
I tried it with the React seed and was successfully able to compile and run JSON Forms without installing Redux :tada: 

However I was not able to run the fallback Redux variant as the `jsonformsReducer` is now missing. Please check whether we can restore this export.

Please also check:
 * All tests should run. Currently the core tests (and maybe more) are failing.
 * All example apps should. Currently at least the React-Material example app doesn't start.

File: packages/core/src/util/type.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+
```
Comment (sdirix) [inline packages/core/src/util/type.ts:25]: Please add a line from where the types were copied.

File: packages/core/src/util/type.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  [Symbol.observable](): Observable<T>
+};
+
+/**
+ * An Observer is used to receive data from an Observable, and is supplied as
+ * an argument to subscribe.
+ */
+export type Observer<T> = {
+  next?(value: T): void
+};
```
Comment (sdirix) [inline packages/core/src/util/type.ts:146]: We don't need these types, right? Please remove them when not needed.

File: packages/core/src/reducers/index.ts [hunk 4]
```diff
@@ -80,8 +79,7 @@ export { JsonFormsCore, ValidationMode };
 
 export const jsonformsReducer = (
```
Comment (sdirix) [inline packages/core/src/reducers/index.ts:80]: Please rename as discussed.

**Review by sdirix — CHANGES_REQUESTED** [review]
Looks good! Some minor comments.

File: packages/core/src/util/type.ts (hunk 3 above)
Comment (TheZoker) [reply packages/core/src/util/type.ts:146]: `Observer` is used by `Observable` and `Observable` is used by `Store`, so I think we need these types here, right?

Comment (sdirix) [reply packages/core/src/util/type.ts:146]: Do we use the Observable of the Store somewhere?

**Review by sdirix — CHANGES_REQUESTED** [review]
Build is failing

**Review by sdirix — CHANGES_REQUESTED** [review]
The typings for the `jsonFormsReducerConfig` fail when trying to use a built version. It seems each of the reducers should declare an appropriate `Reducer<State,Action>` type instead of relying on automatic type conversion.

**Review by sdirix — APPROVED** [review]
LGTM

**Pull request merged**